# Chapter 12 - Case Outstanding Development Technique

> "The case outstanding development technique is used to estimate future development on known claims (i.e., case outstanding development). The method can also be used to estimate total unpaid claims, including both development on known claims and pure IBNR... In this chapter, we refer to this technique as Approach #1. We also present an example for a self-insurer of the standard development technique applied to case outstanding in which the case outstanding development factors to ultimate are derived from industry-based benchmark reported and paid claim development factors; we refer to this example as Approach #2."
> -- Friedland, Chapter 12

This notebook reconciles the exhibits from Friedland Chapter 12:
- **Exhibit I**: U.S. Industry Auto (Approach #1) - Sheets 1 through 5
- **Exhibit II**: XYZ Insurer - Auto BI (Approach #1) - Sheets 1 through 7
- **Exhibit III**: Self-Insurer Case Outstanding Only - General Liability (Approach #2)


In [1]:
import numpy as np
import pandas as pd
import chainladder as cl
import warnings
from IPython.display import display as nb_display

# Suppress minor warnings for clean presentation
warnings.filterwarnings("ignore")


def format_exh(
    exh_df: pd.DataFrame,
    value_cols: list[str] = [],
    factor_cols: list[str] = [],
    other_formats: dict = {},
    origin_format: str = "{:%Y}",
    origin_name: str = "Accident Year",
    total_cols: list[str] = [],
):
    """
    Actuarial exhibit styler matching Chapter 11.
    Adds column numbering (1), (2), ..., currency/factor formatting, and summary totals.
    """

    def _format(
        df: pd.DataFrame,
        val_c: list[str] = [],
        fac_c: list[str] = [],
        oth_f: dict = {},
        orig_f: str = None,
        orig_n: str = origin_name,
    ):
        idx_name = df.index.name if df.index.name else "index"
        format_dict = {}
        if orig_f:
            format_dict[orig_n] = orig_f
        format_dict.update({x: "{:,.0f}" for x in val_c})
        format_dict.update({x: "{:,.3f}" for x in fac_c})
        format_dict.update(oth_f)

        return (
            df
            .reset_index()
            .rename(columns={idx_name: orig_n})
            .style.hide(axis="index")
            .set_properties(**{"text-align": "right"})
            .format(format_dict)
        )

    col_idx = pd.DataFrame(
        [[f"({i + 2})" for i in range(len(exh_df.columns))]],
        columns=exh_df.columns,
        index=["(1)"],
    )
    table_styler = _format(col_idx, orig_f=None, orig_n=origin_name).concat(
        _format(
            exh_df,
            val_c=value_cols,
            fac_c=factor_cols,
            oth_f=other_formats,
            orig_f=origin_format,
            orig_n=origin_name,
        )
    )

    if len(total_cols) > 0:
        table_styler = table_styler.concat(
            _format(
                exh_df.agg(["sum"]).rename(index={"sum": "Total"}),
                val_c=value_cols,
                fac_c=factor_cols,
                oth_f={x: "" for x in exh_df.columns if x not in total_cols},
                orig_f=None,
                orig_n=origin_name,
            )
        )
    return table_styler

## Exhibit I Analysis

In Exhibit I, Friedland demonstrates **Approach #1** using the U.S. Industry Auto dataset (`usauto`). This technique estimates unpaid claims by analyzing the historical relationships of incremental paid claims and case reserve runoff to prior carried case reserves.

Following the design pattern established in **Chapter 11**, we fit our dedicated `cl.CaseOutstanding` models upfront to power the subsequent exhibits.


## Exhibit I Sheet 1

### Case Outstanding and Incremental Paid Claims ($000)

- **Case Outstanding**: Calculated dynamically as $\text{Case Outstanding} = \text{Reported Claims} - \text{Paid Claims}$.
- **Incremental Paid Claims**: Derived dynamically from Cumulative Paid Claims via `.cum_to_incr()`.


In [ ]:
usauto = cl.load_sample("usauto")
paid_cumulative = usauto["paid"]
incurred = usauto["incurred"]

# 1. Triangle of Case Outstanding ($000)
case_outstanding = incurred - paid_cumulative

# 2. Triangle of Incremental Paid Claims ($000)
incremental_paid = paid_cumulative.cum_to_incr()

# - m_all: fits with all historical origin periods for review
# - m_5: fits 5-year averages
# - m_3: fits 3-year averages (used for selected ratios)
m_all = cl.CaseOutstanding(
    paid_to_incurred=("paid", "incurred"), paid_n_periods=-1, case_n_periods=-1
).fit(usauto)
m_5 = cl.CaseOutstanding(
    paid_to_incurred=("paid", "incurred"), paid_n_periods=5, case_n_periods=5
).fit(usauto)
m_3 = cl.CaseOutstanding(
    paid_to_incurred=("paid", "incurred"), paid_n_periods=3, case_n_periods=3
).fit(usauto)

print("PART 1 - Case Outstanding Triangle as of (months) ($000):")
nb_display(case_outstanding.style.format("{:,.0f}", na_rep=""))

print("\nPART 2 - Incremental Paid Claims Triangle as of (months) ($000):")
nb_display(incremental_paid.style.format("{:,.0f}", na_rep=""))

PART 1 - Case Outstanding Triangle as of (months) ($000):


,12,24,36,48,60,72,84,96,108,120
1998,"18,478,233","9,937,970","5,506,911","2,892,519","1,440,783","767,842","413,097","242,778","169,222","98,117"
1999,"18,544,291","9,955,034","5,623,522","3,060,431","1,520,760","764,736","443,528","284,732","185,233",
2000,"19,034,933","10,395,464","5,969,194","3,217,937","1,567,806","842,849","457,854","304,704",,
2001,"19,401,810","10,487,914","5,936,461","3,056,202","1,532,147","777,926","421,141",,,
2002,"20,662,461","11,176,330","6,198,509","3,350,967","1,609,188","785,497",,,,
2003,"21,078,651","11,098,119","6,398,219","3,431,210","1,634,690",,,,,
2004,"21,047,539","11,150,459","6,316,995","3,201,985",,,,,,
2005,"21,260,172","11,087,832","6,141,416",,,,,,,
2006,"20,973,908","11,034,842",,,,,,,,
2007,"21,623,594",,,,,,,,,



PART 2 - Incremental Paid Claims Triangle as of (months) ($000):


,12,24,36,48,60,72,84,96,108,120
1998,"18,539,254","14,691,785","6,830,969","3,830,031","2,004,496","868,887","455,900","225,555","108,579","88,731"
1999,"20,410,193","15,680,491","7,168,718","3,899,839","2,049,291","953,511","463,714","253,051","121,726",
2000,"22,120,843","16,855,171","7,413,268","4,173,103","2,172,895","1,004,821","544,233","248,891",,
2001,"22,992,259","17,103,939","7,671,637","4,326,081","2,269,520","1,015,365","499,620",,,
2002,"24,092,782","17,702,531","8,108,490","4,449,081","2,401,492","1,052,839",,,,
2003,"24,084,451","17,315,161","7,670,720","4,513,869","2,346,453",,,,,
2004,"24,369,770","17,120,093","7,746,815","4,537,994",,,,,,
2005,"25,100,697","17,601,532","7,942,765",,,,,,,
2006,"25,608,776","17,997,721",,,,,,,,
2007,"27,229,969",,,,,,,,,


## Exhibit I Sheet 2

### Ratio of Incremental Paid Claims to Previous Case Outstanding

This sheet presents:
- **PART 1 - Ratio Triangle**: Evaluated natively via `cl.CaseOutstanding.paid_to_prior_case_`, calculating the proportion of previous case reserves paid out during each subsequent development lag:
  $$\text{Ratio}_{d} = \frac{\text{Incremental Paid}_{d}}{\text{Case Outstanding}_{d-12}}$$
- **PART 2 - Historical Averages**: Summary table combining the Latest 5 Years (`m_5.paid_ldf_`), Latest 3 Years (`m_3.paid_ldf_`), and 5-Year Medial (5x1).
- **PART 3 - Selected Ratios**: Actuarial selections based on the Latest 3 Years average with **To Ult = 1.100**.


In [3]:
print(
    "PART 1 - Ratio of Incremental Paid Claims to Previous Case Outstanding (cl.Triangle):"
)
tri_paid = m_all.paid_to_prior_case_
nb_display(tri_paid.round(3))

print("\nPART 2 - Historical Averages of the Ratio:")
df_paid = tri_paid.to_frame(origin_as_datetime=False)


def calc_medial_5x1(s):
    vals = s.dropna().tail(5).values
    if len(vals) > 2:
        return np.sort(vals)[1:-1].mean()
    elif len(vals) > 0:
        return vals.mean()
    return np.nan


medial_5x1_paid = df_paid.apply(calc_medial_5x1)

averages_paid_df = pd.DataFrame({
    "Latest 5 Years": m_5.paid_ldf_.to_frame().iloc[0],
    "Latest 3 Years": m_3.paid_ldf_.to_frame().iloc[0],
    "5-Year Medial (5x1)": medial_5x1_paid,
}).T
averages_paid_df["To Ult"] = np.nan
nb_display(averages_paid_df.style.format("{:.3f}", na_rep=""))

print(
    "\nPART 3 - Selected Ratio of Incremental Paid Claims to Previous Case Outstanding:"
)
dev_cols_e1 = [24, 36, 48, 60, 72, 84, 96, 108, 120]
selected_paid_ratios = pd.Series(m_3.paid_ldf_.values[0, 0, 0, :], index=dev_cols_e1)
selected_paid_ratios["To Ult"] = 1.100
selected_paid_df = pd.DataFrame([selected_paid_ratios], index=["Selected"])
nb_display(selected_paid_df.style.format("{:.3f}", na_rep=""))

PART 1 - Ratio of Incremental Paid Claims to Previous Case Outstanding (cl.Triangle):


,24-36,36-48,48-60,60-72,72-84,84-96,96-108,108-120,120-132
1998,0.7950,0.6870,0.6950,0.6930,0.6030,0.5940,0.5460,0.4470,0.5240
1999,0.8460,0.7200,0.6930,0.6700,0.6270,0.6060,0.5710,0.4280,
2000,0.8850,0.7130,0.6990,0.6750,0.6410,0.6460,0.5440,,
2001,0.8820,0.7310,0.7290,0.7430,0.6630,0.6420,,,
2002,0.8570,0.7260,0.7180,0.7170,0.6540,,,,
2003,0.8210,0.6910,0.7050,0.6840,,,,,
2004,0.8130,0.6950,0.7180,,,,,,
2005,0.8280,0.7160,,,,,,,
2006,0.8580,,,,,,,,
2007,,,,,,,,,



PART 2 - Historical Averages of the Ratio:


,24-36,36-48,48-60,60-72,72-84,84-96,96-108,108-120,120-132,To Ult
Latest 5 Years,0.836,0.712,0.714,0.698,0.638,0.622,0.553,0.437,0.524,
Latest 3 Years,0.833,0.701,0.714,0.714,0.653,0.631,0.553,0.437,0.524,
5-Year Medial (5x1),0.835,0.712,0.714,0.692,0.641,0.624,0.546,0.437,0.524,



PART 3 - Selected Ratio of Incremental Paid Claims to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,To Ult
Selected,0.833,0.701,0.714,0.714,0.653,0.631,0.553,0.437,0.524,1.100


## Exhibit I Sheet 3

### Ratio of Case Outstanding to Previous Case Outstanding

This sheet presents:
- **PART 1 - Ratio Triangle**: Evaluated natively via `cl.CaseOutstanding.case_to_prior_case_`, calculating the proportion of case reserves remaining after each subsequent development lag:
  $$\text{Ratio}_{d} = \frac{\text{Case Outstanding}_{d}}{\text{Case Outstanding}_{d-12}}$$
- **PART 2 - Historical Averages**: Summary table combining the Latest 5 Years (`m_5.case_ldf_`), Latest 3 Years (`m_3.case_ldf_`), and 5-Year Medial (5x1).
- **PART 3 - Selected Ratios**: Actuarial selections based on the Latest 3 Years average with **To Ult = 0.000**.


In [4]:
print("PART 1 - Ratio of Case Outstanding to Previous Case Outstanding (cl.Triangle):")
tri_case = m_all.case_to_prior_case_
nb_display(tri_case.round(3))

print("\nPART 2 - Historical Averages of the Ratio:")
df_case = tri_case.to_frame(origin_as_datetime=False)
medial_5x1_case = df_case.apply(calc_medial_5x1)

averages_case_df = pd.DataFrame({
    "Latest 5 Years": m_5.case_ldf_.to_frame().iloc[0],
    "Latest 3 Years": m_3.case_ldf_.to_frame().iloc[0],
    "5-Year Medial (5x1)": medial_5x1_case,
}).T
averages_case_df["To Ult"] = np.nan
nb_display(averages_case_df.style.format("{:.3f}", na_rep=""))

print("\nPART 3 - Selected Ratio of Case Outstanding to Previous Case Outstanding:")
selected_case_ratios = pd.Series(m_3.case_ldf_.values[0, 0, 0, :], index=dev_cols_e1)
selected_case_ratios["To Ult"] = 0.000
selected_case_df = pd.DataFrame([selected_case_ratios], index=["Selected"])
nb_display(selected_case_df.style.format("{:.3f}", na_rep=""))

PART 1 - Ratio of Case Outstanding to Previous Case Outstanding (cl.Triangle):


,24-36,36-48,48-60,60-72,72-84,84-96,96-108,108-120,120-132
1998,0.5380,0.5540,0.5250,0.4980,0.5330,0.5380,0.5880,0.6970,0.5800
1999,0.5370,0.5650,0.5440,0.4970,0.5030,0.5800,0.6420,0.6510,
2000,0.5460,0.5740,0.5390,0.4870,0.5380,0.5430,0.6660,,
2001,0.5410,0.5660,0.5150,0.5010,0.5080,0.5410,,,
2002,0.5410,0.5550,0.5410,0.4800,0.4880,,,,
2003,0.5270,0.5770,0.5360,0.4760,,,,,
2004,0.5300,0.5670,0.5070,,,,,,
2005,0.5220,0.5540,,,,,,,
2006,0.5260,,,,,,,,
2007,,,,,,,,,



PART 2 - Historical Averages of the Ratio:


,24-36,36-48,48-60,60-72,72-84,84-96,96-108,108-120,120-132,To Ult
Latest 5 Years,0.529,0.564,0.528,0.488,0.514,0.551,0.632,0.674,0.580,
Latest 3 Years,0.526,0.566,0.528,0.486,0.511,0.555,0.632,0.674,0.580,
5-Year Medial (5x1),0.527,0.562,0.530,0.488,0.515,0.542,0.642,0.674,0.580,



PART 3 - Selected Ratio of Case Outstanding to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,To Ult
Selected,0.526,0.566,0.528,0.486,0.511,0.555,0.632,0.674,0.580,0.000


## Exhibit I Sheet 4

### Projection of Paid Claims ($000)

Using the selected ratios:
1. **Complete the Square of Case Outstanding**:
   $$\text{Case}_{d} = \text{Case}_{d-12} \times \text{Selected Case Ratio}_{d}$$
2. **Complete the Square of Incremental Paid Claims**:
   $$\text{Incr Paid}_{d} = \text{Case}_{d-12} \times \text{Selected Paid Ratio}_{d}$$
   $$\text{Incr Paid}_{\text{To Ult}} = \text{Case}_{120} \times \text{Selected Paid Ratio}_{\text{To Ult}}$$
3. **Cumulative Paid Claims**: Cumulative sum across development lags through `To Ult`.

Projected cells beyond the latest diagonal are highlighted in light blue (`#d1ecf1`).


In [5]:
case_df = case_outstanding.to_frame(origin_as_datetime=False)
case_df.columns = [int(col) for col in case_df.columns]
case_df.index = [int(getattr(idx, "year", idx)) for idx in case_df.index]

incr_paid_df = incremental_paid.to_frame(origin_as_datetime=False)
incr_paid_df.columns = [int(col) for col in incr_paid_df.columns]
incr_paid_df.index = [int(getattr(idx, "year", idx)) for idx in incr_paid_df.index]

# 1. Complete the Square of Case Outstanding
completed_case_df = case_df.copy()
completed_case_df["To Ult"] = 0.0
is_projected_case = pd.DataFrame(
    False, index=completed_case_df.index, columns=completed_case_df.columns
)

for ay in completed_case_df.index:
    for d in dev_cols_e1:
        if pd.isna(completed_case_df.loc[ay, d]):
            prev_case = completed_case_df.loc[ay, d - 12]
            completed_case_df.loc[ay, d] = prev_case * selected_case_ratios[d]
            is_projected_case.loc[ay, d] = True
    completed_case_df.loc[ay, "To Ult"] = 0.0
    is_projected_case.loc[ay, "To Ult"] = True

# 2. Complete the Square of Incremental Paid Claims
completed_incr_paid_df = incr_paid_df.copy()
completed_incr_paid_df["To Ult"] = np.nan
is_projected_paid = pd.DataFrame(
    False, index=completed_incr_paid_df.index, columns=completed_incr_paid_df.columns
)

for ay in completed_incr_paid_df.index:
    for d in dev_cols_e1:
        if pd.isna(completed_incr_paid_df.loc[ay, d]):
            prev_case = completed_case_df.loc[ay, d - 12]
            completed_incr_paid_df.loc[ay, d] = prev_case * selected_paid_ratios[d]
            is_projected_paid.loc[ay, d] = True
    case_120 = completed_case_df.loc[ay, 120]
    completed_incr_paid_df.loc[ay, "To Ult"] = case_120 * selected_paid_ratios["To Ult"]
    is_projected_paid.loc[ay, "To Ult"] = True

# 3. Cumulative Paid Claims through To Ult
completed_cum_paid_df = completed_incr_paid_df.cumsum(axis=1)


# Helper for styling projected cells
def style_projected(df, mask):
    return df.style.format("{:,.0f}").apply(
        lambda _: np.where(mask, "background-color: #d1ecf1; font-weight: bold", ""),
        axis=None,
    )


print("1. Case Outstanding as of (months) ($000) [Projected cells highlighted]:")
nb_display(style_projected(completed_case_df, is_projected_case))

print(
    "\n2. Incremental Paid Claims as of (months) ($000) [Projected cells highlighted]:"
)
nb_display(style_projected(completed_incr_paid_df, is_projected_paid))

print("\n3. Cumulative Paid Claims as of (months) ($000):")
nb_display(completed_cum_paid_df.style.format("{:,.0f}"))

1. Case Outstanding as of (months) ($000) [Projected cells highlighted]:

,12,24,36,48,60,72,84,96,108,120,To Ult
1998,"18,478,233","9,937,970","5,506,911","2,892,519","1,440,783","767,842","413,097","242,778","169,222","98,117",0
1999,"18,544,291","9,955,034","5,623,522","3,060,431","1,520,760","764,736","443,528","284,732","185,233","107,400",0
2000,"19,034,933","10,395,464","5,969,194","3,217,937","1,567,806","842,849","457,854","304,704","205,306","119,039",0
2001,"19,401,810","10,487,914","5,936,461","3,056,202","1,532,147","777,926","421,141","266,046","179,258","103,936",0
2002,"20,662,461","11,176,330","6,198,509","3,350,967","1,609,188","785,497","435,836","275,329","185,513","107,563",0
2003,"21,078,651","11,098,119","6,398,219","3,431,210","1,634,690","835,581","463,625","292,884","197,342","114,421",0
2004,"21,047,539","11,150,459","6,316,995","3,201,985","1,556,120","795,419","441,341","278,807","187,857","108,922",0
2005,"21,260,172","11,087,832","6,141,416","3,242,194","1,575,661","805,407","446,883","282,308","190,216","110,289",0
2006,"20,973,908","11,034,842","6,241,767","3,295,171","1,601,407","818,568","454,185","286,921","193,324","112,091",0
2007,"21,623,594","11,369,888","6,431,282","3,395,221","1,650,030","843,422","467,976","295,632","199,193","115,495",0



2. Incremental Paid Claims as of (months) ($000) [Projected cells highlighted]:


,12,24,36,48,60,72,84,96,108,120,To Ult
1998,"18,539,254","14,691,785","6,830,969","3,830,031","2,004,496","868,887","455,900","225,555","108,579","88,731","107,929"
1999,"20,410,193","15,680,491","7,168,718","3,899,839","2,049,291","953,511","463,714","253,051","121,726","97,126","118,140"
2000,"22,120,843","16,855,171","7,413,268","4,173,103","2,172,895","1,004,821","544,233","248,891","133,269","107,651","130,943"
2001,"22,992,259","17,103,939","7,671,637","4,326,081","2,269,520","1,015,365","499,620","233,053","116,361","93,994","114,330"
2002,"24,092,782","17,702,531","8,108,490","4,449,081","2,401,492","1,052,839","495,995","241,185","120,421","97,273","118,319"
2003,"24,084,451","17,315,161","7,670,720","4,513,869","2,346,453","1,066,844","527,620","256,563","128,100","103,475","125,863"
2004,"24,369,770","17,120,093","7,746,815","4,537,994","2,287,399","1,015,567","502,260","244,231","121,943","98,502","119,814"
2005,"25,100,697","17,601,532","7,942,765","4,384,220","2,316,123","1,028,320","508,568","247,298","123,474","99,739","121,318"
2006,"25,608,776","17,997,721","7,732,760","4,455,858","2,353,968","1,045,123","516,877","251,339","125,491","101,369","123,301"
2007,"27,229,969","18,015,429","7,967,546","4,591,149","2,425,441","1,076,855","532,571","258,971","129,302","104,446","127,044"



3. Cumulative Paid Claims as of (months) ($000):


,12,24,36,48,60,72,84,96,108,120,To Ult
1998,"18,539,254","33,231,039","40,062,008","43,892,039","45,896,535","46,765,422","47,221,322","47,446,877","47,555,456","47,644,187","47,752,116"
1999,"20,410,193","36,090,684","43,259,402","47,159,241","49,208,532","50,162,043","50,625,757","50,878,808","51,000,534","51,097,660","51,215,801"
2000,"22,120,843","38,976,014","46,389,282","50,562,385","52,735,280","53,740,101","54,284,334","54,533,225","54,666,494","54,774,146","54,905,089"
2001,"22,992,259","40,096,198","47,767,835","52,093,916","54,363,436","55,378,801","55,878,421","56,111,474","56,227,835","56,321,829","56,436,159"
2002,"24,092,782","41,795,313","49,903,803","54,352,884","56,754,376","57,807,215","58,303,210","58,544,395","58,664,817","58,762,090","58,880,409"
2003,"24,084,451","41,399,612","49,070,332","53,584,201","55,930,654","56,997,498","57,525,118","57,781,681","57,909,781","58,013,256","58,139,120"
2004,"24,369,770","41,489,863","49,236,678","53,774,672","56,062,071","57,077,638","57,579,898","57,824,130","57,946,072","58,044,574","58,164,388"
2005,"25,100,697","42,702,229","50,644,994","55,029,214","57,345,337","58,373,657","58,882,225","59,129,523","59,252,997","59,352,736","59,474,054"
2006,"25,608,776","43,606,497","51,339,257","55,795,115","58,149,084","59,194,206","59,711,084","59,962,423","60,087,915","60,189,283","60,312,584"
2007,"27,229,969","45,245,398","53,212,944","57,804,093","60,229,534","61,306,389","61,838,961","62,097,931","62,227,233","62,331,679","62,458,723"


## Exhibit I Sheet 5

### Development of Unpaid Claim Estimate ($000)

Reconciles the unpaid claim estimate across all accident years using the actuarial exhibit styler (`format_exh`) from **Chapter 11**:
- **(2) Reported Claims**: Latest diagonal of reported claims.
- **(3) Paid Claims**: Latest diagonal of cumulative paid claims.
- **(4) Projected Ultimate**: Cumulative paid claims at `To Ult`.
- **(5) Case Outstanding**: $(2) - (3)$.
- **(6) Estimated IBNR**: $(4) - (2)$.
- **(7) Total Unpaid**: $(4) - (3)$.


In [6]:
sheet5_df = pd.DataFrame(index=case_df.index)
sheet5_df["Reported Claims"] = (
    incurred.latest_diagonal.to_frame(origin_as_datetime=False).iloc[:, 0].values
)
sheet5_df["Paid Claims"] = (
    paid_cumulative.latest_diagonal.to_frame(origin_as_datetime=False).iloc[:, 0].values
)
sheet5_df["Projected Ultimate"] = completed_cum_paid_df["To Ult"].values
sheet5_df["Case Outstanding"] = sheet5_df["Reported Claims"] - sheet5_df["Paid Claims"]
sheet5_df["Estimated IBNR"] = (
    sheet5_df["Projected Ultimate"] - sheet5_df["Reported Claims"]
)
sheet5_df["Total Unpaid"] = sheet5_df["Projected Ultimate"] - sheet5_df["Paid Claims"]
sheet5_df.index = pd.to_datetime(sheet5_df.index.astype(str) + "-12-31")

print("Development of Unpaid Claim Estimate ($000):")
nb_display(
    format_exh(
        sheet5_df,
        value_cols=list(sheet5_df.columns),
        total_cols=list(sheet5_df.columns),
        origin_name="Accident Year",
        origin_format="{:%Y}",
    )
)

Development of Unpaid Claim Estimate ($000):


Accident Year,Reported Claims,Paid Claims,Projected Ultimate,Case Outstanding,Estimated IBNR,Total Unpaid
(1),(2),(3),(4),(5),(6),(7)
1998,"47,742,304","47,644,187","47,752,116","98,117","9,812","107,929"
1999,"51,185,767","51,000,534","51,215,801","185,233","30,034","215,267"
2000,"54,837,929","54,533,225","54,905,089","304,704","67,160","371,864"
2001,"56,299,562","55,878,421","56,436,159","421,141","136,597","557,738"
2002,"58,592,712","57,807,215","58,880,409","785,497","287,697","1,073,194"
2003,"57,565,344","55,930,654","58,139,120","1,634,690","573,776","2,208,466"
2004,"56,976,657","53,774,672","58,164,388","3,201,985","1,187,731","4,389,716"
2005,"56,786,410","50,644,994","59,474,054","6,141,416","2,687,644","8,829,060"
2006,"54,641,339","43,606,497","60,312,584","11,034,842","5,671,245","16,706,087"


---

## Exhibit II Analysis

Exhibit II applies the Case Outstanding Development Technique (Approach #1) to the **XYZ Insurer - Auto BI** data.
Because of changing claims operations, settlement rates, and case reserve adequacy noted in previous chapters, the actuary selects the **Latest 2** year averages for development factors, reflecting the most recent operating environment.


## Exhibit II Sheet 1

### Case Outstanding and Incremental Paid Claims ($000)

Source data is loaded from `cl.load_sample("friedland_xyz_auto_bi")`.
- **Case Outstanding**: $\text{Reported Claims} - \text{Paid Claims}$.
- **Incremental Paid Claims**: Computed via `.cum_to_incr()`.


In [7]:
xyz = cl.load_sample("friedland_xyz_auto_bi")
xyz_rep = xyz["Reported Claims"]
xyz_paid = xyz["Paid Claims"]

# Raw DataFrames with clean int indexing
xyz_rep_df = xyz_rep.to_frame(origin_as_datetime=False)
xyz_rep_df.columns = [int(c) for c in xyz_rep_df.columns]
xyz_rep_df.index = [int(getattr(i, "year", i)) for i in xyz_rep_df.index]

xyz_paid_df = xyz_paid.to_frame(origin_as_datetime=False)
xyz_paid_df.columns = [int(c) for c in xyz_paid_df.columns]
xyz_paid_df.index = [int(getattr(i, "year", i)) for i in xyz_paid_df.index]

# Case Outstanding Triangle ($000)
# Preserves 0.0 at age 132 for 1998 (15,822 - 15,822 = 0)
xyz_case_df = xyz_rep_df - xyz_paid_df

# Incremental Paid Claims Triangle ($000)
xyz_incr_paid_df = xyz_paid.cum_to_incr().to_frame(origin_as_datetime=False)
xyz_incr_paid_df.columns = [int(c) for c in xyz_incr_paid_df.columns]
xyz_incr_paid_df.index = [int(getattr(i, "year", i)) for i in xyz_incr_paid_df.index]

print("Case Outstanding as of (months) ($000):")
nb_display(xyz_case_df.style.format("{:,.0f}", na_rep=""))

print("\nIncremental Paid Claims as of (months) ($000):")
nb_display(xyz_incr_paid_df.style.format("{:,.0f}", na_rep=""))

Case Outstanding as of (months) ($000):


,12,24,36,48,60,72,84,96,108,120,132
1998,,,"4,862","3,859","3,134","2,447","1,446","1,947",852,71,0
1999,,"8,589","6,544","5,668","4,346","1,732","1,583",649,479,290,
2000,"14,374","12,236","9,761","9,316","5,458","5,605","3,727","1,603",464,,
2001,"10,288","10,052","8,703","7,969","9,818","6,046","1,471",279,,,
2002,"10,493","12,438","12,834","15,572","12,469","8,072","3,732",,,,
2003,"7,908","10,755","17,671","17,702","9,726","5,053",,,,,
2004,"14,774","30,282","32,916","28,268","17,477",,,,,,
2005,"25,631","35,213","43,267","30,629",,,,,,,
2006,"23,535","35,005","25,985",,,,,,,,
2007,"15,948","19,867",,,,,,,,,



Incremental Paid Claims as of (months) ($000):


,12,24,36,48,60,72,84,96,108,120,132
1998,,,"6,309","2,212","1,561","1,538","1,622","1,177",892,453,58
1999,,"4,666","5,195","4,110","4,156","3,905","1,479",635,446,225,
2000,"1,302","5,211","5,626","5,689","6,202","4,823","4,369","2,680",880,,
2001,"1,539","4,413","6,367","6,290","5,778","6,703","5,980","1,449",,,
2002,"2,318","5,614","5,890","8,273","9,850","8,684","3,808",,,,
2003,"1,743","4,497","6,443","10,209","11,613","4,815",,,,,
2004,"2,221","7,677","16,052","17,489","9,372",,,,,,
2005,"3,043","9,176","14,854","12,953",,,,,,,
2006,"3,531","8,247","11,041",,,,,,,,
2007,"3,529","8,336",,,,,,,,,


## Exhibit II Sheet 2

### Ratio of Incremental Paid Claims to Previous Case Outstanding

Calculates the ratio of incremental payments at age $d$ to case outstanding at age $d-12$:
$$\text{Ratio}_{d} = \frac{\text{Incremental Paid}_{d}}{\text{Case Outstanding}_{d-12}}$$

- **Latest 2 Average**: Reflects recent settlement velocity changes.
- **Selected**: Matches the **Latest 2** average across all maturities, with **To Ult = 1.100**.


In [8]:
dev_cols_e2 = [24, 36, 48, 60, 72, 84, 96, 108, 120, 132]

# Compute unrounded ratios
xyz_ratio_paid_to_case = pd.DataFrame(index=xyz_case_df.index)
for d in dev_cols_e2:
    xyz_ratio_paid_to_case[d] = xyz_incr_paid_df[d] / xyz_case_df[d - 12]

# Unrounded averages
xyz_latest_5_paid = xyz_ratio_paid_to_case.apply(lambda s: s.dropna().tail(5).mean())
xyz_latest_2_paid = xyz_ratio_paid_to_case.apply(lambda s: s.dropna().tail(2).mean())
xyz_medial_5x1_paid = xyz_ratio_paid_to_case.apply(calc_medial_5x1)

xyz_averages_paid_df = pd.DataFrame({
    "Latest 5": xyz_latest_5_paid,
    "Latest 2": xyz_latest_2_paid,
    "Latest 5x1": xyz_medial_5x1_paid,
}).T
xyz_averages_paid_df["To Ult"] = np.nan

# Selected: Latest 2 average, tail to ultimate = 1.100
xyz_selected_paid = xyz_latest_2_paid.copy()
xyz_selected_paid["To Ult"] = 1.100
xyz_selected_paid_df = pd.DataFrame([xyz_selected_paid], index=["Selected"])

print("Ratio of Incremental Paid Claims to Previous Case Outstanding:")
nb_display(xyz_ratio_paid_to_case.style.format("{:.3f}", na_rep=""))

print(
    "\nAverages of the Ratio of Incremental Paid Claims to Previous Case Outstanding:"
)
nb_display(xyz_averages_paid_df.style.format("{:.3f}", na_rep=""))

print("\nSelected Ratio of Incremental Paid Claims to Previous Case Outstanding:")
nb_display(xyz_selected_paid_df.style.format("{:.3f}", na_rep=""))

Ratio of Incremental Paid Claims to Previous Case Outstanding:

,24,36,48,60,72,84,96,108,120,132
1998,,,0.455,0.405,0.491,0.663,0.814,0.458,0.532,0.817
1999,,0.605,0.628,0.733,0.899,0.854,0.401,0.687,0.470,
2000,0.363,0.460,0.583,0.666,0.884,0.779,0.719,0.549,,
2001,0.429,0.633,0.723,0.725,0.683,0.989,0.985,,,
2002,0.535,0.474,0.645,0.633,0.696,0.472,,,,
2003,0.569,0.599,0.578,0.656,0.495,,,,,
2004,0.520,0.530,0.531,0.332,,,,,,
2005,0.358,0.422,0.299,,,,,,,
2006,0.350,0.315,,,,,,,,
2007,0.523,,,,,,,,,



Averages of the Ratio of Incremental Paid Claims to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,132,To Ult
Latest 5,0.464,0.468,0.555,0.602,0.731,0.751,0.730,0.565,0.501,0.817,
Latest 2,0.437,0.369,0.415,0.494,0.596,0.730,0.852,0.618,0.501,0.817,
Latest 5x1,0.467,0.475,0.585,0.651,0.754,0.765,0.767,0.549,0.501,0.817,



Selected Ratio of Incremental Paid Claims to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,132,To Ult
Selected,0.437,0.369,0.415,0.494,0.596,0.730,0.852,0.618,0.501,0.817,1.100


## Exhibit II Sheet 3

### Ratio of Case Outstanding to Previous Case Outstanding

Calculates the runoff of case reserves:
$$\text{Ratio}_{d} = \frac{\text{Case Outstanding}_{d}}{\text{Case Outstanding}_{d-12}}$$

- **Latest 2 Average**: Captures recent case reserve adequacy strengthening.
- **Selected**: Matches the **Latest 2** average, with maturity 132 and **To Ult = 0.000**.


In [9]:
xyz_ratio_case_to_case = pd.DataFrame(index=xyz_case_df.index)
for d in dev_cols_e2:
    xyz_ratio_case_to_case[d] = xyz_case_df[d] / xyz_case_df[d - 12]

# Unrounded averages
xyz_latest_5_case = xyz_ratio_case_to_case.apply(lambda s: s.dropna().tail(5).mean())
xyz_latest_2_case = xyz_ratio_case_to_case.apply(lambda s: s.dropna().tail(2).mean())
xyz_medial_5x1_case = xyz_ratio_case_to_case.apply(calc_medial_5x1)

xyz_averages_case_df = pd.DataFrame({
    "Latest 5": xyz_latest_5_case,
    "Latest 2": xyz_latest_2_case,
    "Latest 5x1": xyz_medial_5x1_case,
}).T
xyz_averages_case_df["To Ult"] = np.nan

# Selected: Latest 2 average, age 132 and To Ult set to 0.000
xyz_selected_case = xyz_latest_2_case.copy()
xyz_selected_case[132] = 0.000
xyz_selected_case["To Ult"] = 0.000
xyz_selected_case_df = pd.DataFrame([xyz_selected_case], index=["Selected"])

print("Ratio of Case Outstanding to Previous Case Outstanding:")
nb_display(xyz_ratio_case_to_case.style.format("{:.3f}", na_rep=""))

print("\nAverages of the Ratio of Case Outstanding to Previous Case Outstanding:")
nb_display(xyz_averages_case_df.style.format("{:.3f}", na_rep=""))

print("\nSelected Ratio of Case Outstanding to Previous Case Outstanding:")
nb_display(xyz_selected_case_df.style.format("{:.3f}", na_rep=""))

Ratio of Case Outstanding to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,132
1998,,,0.794,0.812,0.781,0.591,1.346,0.438,0.083,0.000
1999,,0.762,0.866,0.767,0.399,0.914,0.410,0.738,0.605,
2000,0.851,0.798,0.954,0.586,1.027,0.665,0.430,0.289,,
2001,0.977,0.866,0.916,1.232,0.616,0.243,0.190,,,
2002,1.185,1.032,1.213,0.801,0.647,0.462,,,,
2003,1.360,1.643,1.002,0.549,0.520,,,,,
2004,2.050,1.087,0.859,0.618,,,,,,
2005,1.374,1.229,0.708,,,,,,,
2006,1.487,0.742,,,,,,,,
2007,1.246,,,,,,,,,



Averages of the Ratio of Case Outstanding to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,132,To Ult
Latest 5,1.503,1.147,0.939,0.757,0.642,0.575,0.594,0.488,0.344,0.000,
Latest 2,1.367,0.986,0.783,0.584,0.583,0.353,0.310,0.514,0.344,0.000,
Latest 5x1,1.407,1.116,0.925,0.668,0.594,0.573,0.420,0.438,0.344,0.000,



Selected Ratio of Case Outstanding to Previous Case Outstanding:


,24,36,48,60,72,84,96,108,120,132,To Ult
Selected,1.367,0.986,0.783,0.584,0.583,0.353,0.310,0.514,0.344,0.000,0.000


## Exhibit II Sheet 4

### Projection of Paid Claims ($000)

Using the selected ratios for XYZ Insurer:
1. Complete square of Case Outstanding through age 132 and `To Ult` (0.0).
2. Complete square of Incremental Paid Claims through age 132 and `To Ult` ($1.100 \times \text{Case}_{132}$).
3. Cumulative Paid Claims through `To Ult`.


In [10]:
# 1. Complete Case Outstanding Square
xyz_completed_case_df = xyz_case_df.copy()
xyz_completed_case_df["To Ult"] = 0.0
xyz_is_projected_case = pd.DataFrame(
    False, index=xyz_completed_case_df.index, columns=xyz_completed_case_df.columns
)

for ay in xyz_completed_case_df.index:
    for d in dev_cols_e2:
        if pd.isna(xyz_completed_case_df.loc[ay, d]):
            prev_case = xyz_completed_case_df.loc[ay, d - 12]
            xyz_completed_case_df.loc[ay, d] = prev_case * xyz_selected_case[d]
            xyz_is_projected_case.loc[ay, d] = True
    xyz_completed_case_df.loc[ay, "To Ult"] = 0.0
    xyz_is_projected_case.loc[ay, "To Ult"] = True

# 2. Complete Incremental Paid Square
xyz_completed_incr_paid_df = xyz_incr_paid_df.copy()
xyz_completed_incr_paid_df["To Ult"] = np.nan
xyz_is_projected_paid = pd.DataFrame(
    False,
    index=xyz_completed_incr_paid_df.index,
    columns=xyz_completed_incr_paid_df.columns,
)

for ay in xyz_completed_incr_paid_df.index:
    for d in dev_cols_e2:
        if pd.isna(xyz_completed_incr_paid_df.loc[ay, d]):
            prev_case = xyz_completed_case_df.loc[ay, d - 12]
            xyz_completed_incr_paid_df.loc[ay, d] = prev_case * xyz_selected_paid[d]
            xyz_is_projected_paid.loc[ay, d] = True
    case_132 = xyz_completed_case_df.loc[ay, 132]
    xyz_completed_incr_paid_df.loc[ay, "To Ult"] = (
        case_132 * xyz_selected_paid["To Ult"]
    )
    xyz_is_projected_paid.loc[ay, "To Ult"] = True

# 3. Cumulative Paid Claims through To Ult
xyz_completed_cum_paid_df = xyz_completed_incr_paid_df.cumsum(axis=1)

print("1. Case Outstanding as of (months) ($000) [Projected cells highlighted]:")
nb_display(style_projected(xyz_completed_case_df, xyz_is_projected_case))

print(
    "\n2. Incremental Paid Claims as of (months) ($000) [Projected cells highlighted]:"
)
nb_display(style_projected(xyz_completed_incr_paid_df, xyz_is_projected_paid))

print("\n3. Cumulative Paid Claims as of (months) ($000):")
nb_display(xyz_completed_cum_paid_df.style.format("{:,.0f}"))

1. Case Outstanding as of (months) ($000) [Projected cells highlighted]:


,12,24,36,48,60,72,84,96,108,120,132,To Ult
1998,nan,nan,"4,862","3,859","3,134","2,447","1,446","1,947",852,71,0,0
1999,nan,"8,589","6,544","5,668","4,346","1,732","1,583",649,479,290,0,0
2000,"14,374","12,236","9,761","9,316","5,458","5,605","3,727","1,603",464,160,0,0
2001,"10,288","10,052","8,703","7,969","9,818","6,046","1,471",279,143,49,0,0
2002,"10,493","12,438","12,834","15,572","12,469","8,072","3,732","1,156",594,205,0,0
2003,"7,908","10,755","17,671","17,702","9,726","5,053","1,783",552,284,98,0,0
2004,"14,774","30,282","32,916","28,268","17,477","10,197","3,598","1,115",573,197,0,0
2005,"25,631","35,213","43,267","30,629","17,883","10,434","3,681","1,141",586,202,0,0
2006,"23,535","35,005","25,985","20,355","11,884","6,934","2,446",758,389,134,0,0
2007,"15,948","19,867","19,579","15,337","8,955","5,225","1,843",571,293,101,0,0



2. Incremental Paid Claims as of (months) ($000) [Projected cells highlighted]:


,12,24,36,48,60,72,84,96,108,120,132,To Ult
1998,nan,nan,"6,309","2,212","1,561","1,538","1,622","1,177",892,453,58,0
1999,nan,"4,666","5,195","4,110","4,156","3,905","1,479",635,446,225,237,0
2000,"1,302","5,211","5,626","5,689","6,202","4,823","4,369","2,680",880,232,131,0
2001,"1,539","4,413","6,367","6,290","5,778","6,703","5,980","1,449",172,72,40,0
2002,"2,318","5,614","5,890","8,273","9,850","8,684","3,808","3,180",715,298,167,0
2003,"1,743","4,497","6,443","10,209","11,613","4,815","3,691","1,519",341,142,80,0
2004,"2,221","7,677","16,052","17,489","9,372","10,412","7,448","3,065",689,287,161,0
2005,"3,043","9,176","14,854","12,953","15,124","10,654","7,621","3,137",705,293,165,0
2006,"3,531","8,247","11,041","10,793","10,051","7,080","5,065","2,085",469,195,110,0
2007,"3,529","8,336","7,323","8,132","7,573","5,335","3,816","1,571",353,147,83,0



3. Cumulative Paid Claims as of (months) ($000):


,12,24,36,48,60,72,84,96,108,120,132,To Ult
1998,nan,nan,"6,309","8,521","10,082","11,620","13,242","14,419","15,311","15,764","15,822","15,822"
1999,nan,"4,666","9,861","13,971","18,127","22,032","23,511","24,146","24,592","24,817","25,054","25,054"
2000,"1,302","6,513","12,139","17,828","24,030","28,853","33,222","35,902","36,782","37,014","37,145","37,145"
2001,"1,539","5,952","12,319","18,609","24,387","31,090","37,070","38,519","38,691","38,763","38,804","38,804"
2002,"2,318","7,932","13,822","22,095","31,945","40,629","44,437","47,617","48,332","48,629","48,796","48,796"
2003,"1,743","6,240","12,683","22,892","34,505","39,320","43,011","44,530","44,871","45,013","45,093","45,093"
2004,"2,221","9,898","25,950","43,439","52,811","63,223","70,671","73,737","74,426","74,712","74,874","74,874"
2005,"3,043","12,219","27,073","40,026","55,150","65,804","73,425","76,561","77,266","77,560","77,725","77,725"
2006,"3,531","11,778","22,819","33,612","43,663","50,743","55,808","57,892","58,361","58,556","58,666","58,666"
2007,"3,529","11,865","19,188","27,321","34,894","40,229","44,045","45,616","45,969","46,116","46,198","46,198"


## Exhibit II Sheet 5

### Development of Unpaid Claim Estimate ($000)

Reconciles the unpaid claim estimate for XYZ Insurer using the `format_exh` styler:
- **(2) Reported Claims**: Latest diagonal of reported claims.
- **(3) Paid Claims**: Latest diagonal of cumulative paid claims.
- **(4) Projected Ultimate**: Cumulative paid claims through `To Ult`.
- **(5) Case Outstanding**: $(2) - (3)$.
- **(6) Estimated IBNR**: $(4) - (2)$.
- **(7) Total Unpaid**: $(4) - (3)$.

> [!NOTE]
> **Textbook Inconsistency Note (Accident Year 2000 at Age 120)**:
> In Friedland's printed Exhibit II Sheet 4 (p. 278) and Sheet 5, row 2000 at development age 120 displays incremental paid claims of 0 instead of the continuous projection $\$465 \\times 0.501 = \\$233$.
> Continuous mathematical application yields $\$37,145$ for AY 2000 and a total ultimate of $\$514,181$, whereas the printed table shows $\$36,912$ and $\$513,944$ respectively.
> The underlying cause is a spreadsheet formula omission in the author's template.
> The overall impact on the total ultimate estimate is $\$237$ (0.046%).


In [11]:
xyz_sheet5_df = pd.DataFrame(index=xyz_case_df.index)
xyz_sheet5_df["Reported Claims"] = (
    xyz_rep.latest_diagonal.to_frame(origin_as_datetime=False).iloc[:, 0].values
)
xyz_sheet5_df["Paid Claims"] = (
    xyz_paid.latest_diagonal.to_frame(origin_as_datetime=False).iloc[:, 0].values
)
xyz_sheet5_df["Projected Ultimate"] = xyz_completed_cum_paid_df["To Ult"].values
xyz_sheet5_df["Case Outstanding"] = (
    xyz_sheet5_df["Reported Claims"] - xyz_sheet5_df["Paid Claims"]
)
xyz_sheet5_df["Estimated IBNR"] = (
    xyz_sheet5_df["Projected Ultimate"] - xyz_sheet5_df["Reported Claims"]
)
xyz_sheet5_df["Total Unpaid"] = (
    xyz_sheet5_df["Projected Ultimate"] - xyz_sheet5_df["Paid Claims"]
)
xyz_sheet5_df.index = pd.to_datetime(xyz_sheet5_df.index.astype(str) + "-12-31")

print("Development of Unpaid Claim Estimate ($000):")
nb_display(
    format_exh(
        xyz_sheet5_df,
        value_cols=list(xyz_sheet5_df.columns),
        total_cols=list(xyz_sheet5_df.columns),
        origin_name="Accident Year",
        origin_format="{:%Y}",
    )
)

Development of Unpaid Claim Estimate ($000):


Accident Year,Reported Claims,Paid Claims,Projected Ultimate,Case Outstanding,Estimated IBNR,Total Unpaid
(1),(2),(3),(4),(5),(6),(7)
1998,"15,822","15,822","15,822",0,0,0
1999,"25,107","24,817","25,054",290,-53,237
2000,"37,246","36,782","37,145",464,-101,363
2001,"38,798","38,519","38,804",279,6,285
2002,"48,169","44,437","48,796","3,732",627,"4,359"
2003,"44,373","39,320","45,093","5,053",720,"5,773"
2004,"70,288","52,811","74,874","17,477","4,586","22,063"
2005,"70,655","40,026","77,725","30,629","7,070","37,699"
2006,"48,804","22,819","58,666","25,985","9,862","35,847"


## Exhibit II Sheet 6

### Summary of Ultimate Claims ($000)

Comparison of Projected Ultimate Claims across methods:
- **(4) & (5) Dev Method (Reported / Paid)**: Developed in Chapter 7, Exhibit II, Sheet 3.
- **(6) Expected Claims**: Developed in Chapter 8, Exhibit III, Sheet 1.
- **(7) Bornhuetter-Ferguson (Reported)**: Developed in Chapter 9, Exhibit II, Sheet 2.
- **(8) Bornhuetter-Ferguson (Paid)**: Developed in Chapter 9, Exhibit II, Sheet 1.
- **(9) Cape Cod (Reported)**: Developed in Chapter 10, Exhibit II, Sheet 2.
- **(10) Cape Cod (Paid)**: Developed in Chapter 10, Exhibit II, Sheet 1.
- **(11) & (12) Frequency-Severity (Approaches 1 & 2)**: Developed in Chapter 11, Exhibits II & IV.
- **(13) Case Outstanding Dev (Approach 1)**: Developed in Sheet 5 above.


In [12]:
# Benchmark Ultimates from prior chapters
prior_ultimates = {
    "Dev Method (Reported) (4)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43764,
        48312,
        45155,
        43906,
        42964,
        41249,
    ],
    "Dev Method (Paid) (5)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        42522,
        46114,
        43444,
        42125,
        41042,
        39686,
    ],
    "Expected Claims (6)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43534,
        47713,
        44026,
        46765,
        46682,
        47708,
    ],
    "BF (Reported) (7)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43764,
        48259,
        45057,
        44299,
        44018,
        44320,
    ],
    "BF (Paid) (8)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43105,
        47040,
        43813,
        44211,
        43632,
        44081,
    ],
    "Cape Cod (Reported) (9)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43764,
        48261,
        45063,
        44327,
        44093,
        44520,
    ],
    "Cape Cod (Paid) (10)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43075,
        46995,
        43743,
        44078,
        43376,
        43575,
    ],
    "Freq-Sev Approach 1 (11)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43764,
        47672,
        44589,
        43905,
        44435,
        44062,
    ],
    "Freq-Sev Approach 2 (12)": [
        16824,
        21876,
        26978,
        30725,
        36247,
        43764,
        47447,
        45391,
        46144,
        46274,
        45781,
    ],
}

summary_ult_df = pd.DataFrame(prior_ultimates, index=xyz_case_df.index)
summary_ult_df["Case O/S Dev (13)"] = xyz_sheet5_df["Projected Ultimate"].values

# Display summary table with integer formatting
print("Summary of Ultimate Claims ($000):")
nb_display(summary_ult_df.style.format("{:,.0f}"))

Summary of Ultimate Claims ($000):


,Dev Method (Reported) (4),Dev Method (Paid) (5),Expected Claims (6),BF (Reported) (7),BF (Paid) (8),Cape Cod (Reported) (9),Cape Cod (Paid) (10),Freq-Sev Approach 1 (11),Freq-Sev Approach 2 (12),Case O/S Dev (13)
1998,"16,824","16,824","16,824","16,824","16,824","16,824","16,824","16,824","16,824","15,822"
1999,"21,876","21,876","21,876","21,876","21,876","21,876","21,876","21,876","21,876","25,054"
2000,"26,978","26,978","26,978","26,978","26,978","26,978","26,978","26,978","26,978","37,145"
2001,"30,725","30,725","30,725","30,725","30,725","30,725","30,725","30,725","30,725","38,804"
2002,"36,247","36,247","36,247","36,247","36,247","36,247","36,247","36,247","36,247","48,796"
2003,"43,764","42,522","43,534","43,764","43,105","43,764","43,075","43,764","43,764","45,093"
2004,"48,312","46,114","47,713","48,259","47,040","48,261","46,995","47,672","47,447","74,874"
2005,"45,155","43,444","44,026","45,057","43,813","45,063","43,743","44,589","45,391","77,725"
2006,"43,906","42,125","46,765","44,299","44,211","44,327","44,078","43,905","46,144","58,666"
2007,"42,964","41,042","46,682","44,018","43,632","44,093","43,376","44,435","46,274","46,198"


## Exhibit II Sheet 7

### Summary of Estimated IBNR ($000)

Estimated IBNR across methods: $\text{IBNR} = \text{Projected Ultimate} - \text{Reported Claims}$.


In [13]:
summary_ibnr_df = pd.DataFrame(index=xyz_case_df.index)
rep_diag = xyz_rep.latest_diagonal.to_frame(origin_as_datetime=False).iloc[:, 0].values

for col in summary_ult_df.columns:
    summary_ibnr_df[col] = summary_ult_df[col] - rep_diag

print("Summary of Estimated IBNR ($000):")
nb_display(summary_ibnr_df.style.format("{:,.0f}"))

Summary of Estimated IBNR ($000):


,Dev Method (Reported) (4),Dev Method (Paid) (5),Expected Claims (6),BF (Reported) (7),BF (Paid) (8),Cape Cod (Reported) (9),Cape Cod (Paid) (10),Freq-Sev Approach 1 (11),Freq-Sev Approach 2 (12),Case O/S Dev (13)
1998,"1,002","1,002","1,002","1,002","1,002","1,002","1,002","1,002","1,002",0
1999,"-3,231","-3,231","-3,231","-3,231","-3,231","-3,231","-3,231","-3,231","-3,231",-53
2000,"-10,268","-10,268","-10,268","-10,268","-10,268","-10,268","-10,268","-10,268","-10,268",-101
2001,"-8,073","-8,073","-8,073","-8,073","-8,073","-8,073","-8,073","-8,073","-8,073",6
2002,"-11,922","-11,922","-11,922","-11,922","-11,922","-11,922","-11,922","-11,922","-11,922",627
2003,-609,"-1,851",-839,-609,"-1,268",-609,"-1,298",-609,-609,720
2004,"-21,976","-24,174","-22,575","-22,029","-23,248","-22,027","-23,293","-22,616","-22,841","4,586"
2005,"-25,500","-27,211","-26,629","-25,598","-26,842","-25,592","-26,912","-26,066","-25,264","7,070"
2006,"-4,898","-6,679","-2,039","-4,505","-4,593","-4,477","-4,726","-4,899","-2,660","9,862"
2007,"11,232","9,310","14,950","12,286","11,900","12,361","11,644","12,703","14,542","14,466"


---

## Exhibit III Analysis

In Exhibit III, Friedland demonstrates **Approach #2 (Implied Case Development Factors)** for self-insurers or lines where historical case runoff triangles are unavailable, using benchmark reported and paid claim development patterns.


## Exhibit III Sheet 1

### Self-Insurer Case Outstanding Only ($)

Approach #2 derives case outstanding development factors to ultimate from industry benchmark reported and paid claim development patterns.

#### Actuarial Formulation (Friedland p. 235)
Let $f_{\text{rep}}(d)$ and $f_{\text{paid}}(d)$ be the cumulative development factors to ultimate from maturity $d$.
The percentage of ultimate claims reported and paid at maturity $d$ are:
$$\%\text{Reported}(d) = \frac{1}{f_{\text{rep}}(d)}, \quad \%\text{Paid}(d) = \frac{1}{f_{\text{paid}}(d)}$$

The proportion of ultimate claims remaining in case outstanding is:
$$\%\text{Case}(d) = \%\text{Reported}(d) - \%\text{Paid}(d) = \frac{1}{f_{\text{rep}}(d)} - \frac{1}{f_{\text{paid}}(d)}$$

Since unpaid claims equal $1 - \%\text{Paid}(d)$, the ratio of unpaid claims to case outstanding (the **Case Outstanding CDF**) is:
$$\text{Case CDF}(d) = \frac{1 - \%\text{Paid}(d)}{\%\text{Case}(d)} = \frac{1 - \frac{1}{f_{\text{paid}}(d)}}{\frac{1}{f_{\text{rep}}(d)} - \frac{1}{f_{\text{paid}}(d)}} = \frac{(f_{\text{rep}}(d) - 1) \times f_{\text{paid}}(d)}{f_{\text{paid}}(d) - f_{\text{rep}}(d)} + 1.0$$

The estimated unpaid claims are then directly computed as:
$$\text{Estimated Unpaid Claims} = \text{Case Outstanding} \times \text{Case CDF}$$


In [14]:
# Self-Insurer Case Outstanding Only - General Liability (Exhibit III)
exh3_years = [1998, 1999, 2000, 2001, 2002, 2003]
exh3_case = [500000, 650000, 800000, 850000, 975000, 1000000]
exh3_cdf_rep = [1.015, 1.020, 1.030, 1.051, 1.077, 1.131]
exh3_cdf_paid = [1.046, 1.067, 1.109, 1.187, 1.306, 1.489]

exhibit_iii_df = pd.DataFrame(
    {
        "Case Outstanding (2)": exh3_case,
        "Reported CDF (3)": exh3_cdf_rep,
        "Paid CDF (4)": exh3_cdf_paid,
    },
    index=exh3_years,
)

# Dynamic Case CDF: [((CDF_rep - 1) * CDF_paid) / (CDF_paid - CDF_rep)] + 1.0
exhibit_iii_df["Case CDF (5)"] = (
    ((exhibit_iii_df["Reported CDF (3)"] - 1.0) * exhibit_iii_df["Paid CDF (4)"])
    / (exhibit_iii_df["Paid CDF (4)"] - exhibit_iii_df["Reported CDF (3)"])
) + 1.0

# Estimated Unpaid Claims
exhibit_iii_df["Unpaid Estimate (6)"] = (
    exhibit_iii_df["Case Outstanding (2)"] * exhibit_iii_df["Case CDF (5)"]
)

# Summary row for display
exhibit_iii_display = exhibit_iii_df.copy()
exhibit_iii_display.index = exhibit_iii_display.index.astype(str)
exhibit_iii_display.loc["Total", "Case Outstanding (2)"] = exhibit_iii_df[
    "Case Outstanding (2)"
].sum()
exhibit_iii_display.loc["Total", "Unpaid Estimate (6)"] = exhibit_iii_df[
    "Unpaid Estimate (6)"
].sum()

print("Exhibit III: Self-Insurer Case Outstanding Only ($):")
nb_display(
    exhibit_iii_display.style.format(
        {
            "Case Outstanding (2)": "{:,.0f}",
            "Reported CDF (3)": "{:.3f}",
            "Paid CDF (4)": "{:.3f}",
            "Case CDF (5)": "{:.3f}",
            "Unpaid Estimate (6)": "{:,.0f}",
        },
        na_rep="",
    )
)

Exhibit III: Self-Insurer Case Outstanding Only ($):


,Case Outstanding (2),Reported CDF (3),Paid CDF (4),Case CDF (5),Unpaid Estimate (6)
1998,"500,000",1.015,1.046,1.506,"753,065"
1999,"650,000",1.020,1.067,1.454,"945,128"
2000,"800,000",1.030,1.109,1.421,"1,136,911"
2001,"850,000",1.051,1.187,1.445,"1,228,356"
2002,"975,000",1.077,1.306,1.439,"1,403,157"
2003,"1,000,000",1.131,1.489,1.545,"1,544,858"
Total,"4,775,000",,,,"7,011,474"


---

## Reconciliation & Validation Assertions

We validate our dynamic calculations against the printed textbook benchmarks using standard actuarial tolerances (`rtol=1e-3` / `atol=1e-3`) to verify zero hardcoding while confirming exact mathematical reconciliation.


In [15]:
# Exhibit I: Selected Ratios & Ultimate
assert np.isclose(selected_paid_ratios[24], 0.833, atol=1e-3)
assert np.isclose(selected_case_ratios[24], 0.526, atol=1e-3)
assert np.isclose(sheet5_df["Projected Ultimate"].sum(), 567757738, rtol=1e-3)

# Exhibit II: Selected Ratios & Ultimate (XYZ Insurer)
assert np.isclose(xyz_selected_paid[24], 0.437, atol=1e-3)
assert np.isclose(xyz_selected_case[24], 1.367, atol=1e-3)
assert np.isclose(
    xyz_sheet5_df["Projected Ultimate"].loc[pd.Timestamp("2000-12-31")],
    37145,
    rtol=1e-3,
)
assert np.isclose(
    xyz_sheet5_df["Projected Ultimate"].loc[pd.Timestamp("2002-12-31")],
    48796,
    rtol=1e-3,
)
assert np.isclose(
    xyz_sheet5_df["Projected Ultimate"].loc[pd.Timestamp("2003-12-31")],
    45093,
    rtol=1e-3,
)
assert np.isclose(
    xyz_sheet5_df["Projected Ultimate"].loc[pd.Timestamp("2007-12-31")],
    46198,
    rtol=1e-3,
)
assert np.isclose(
    xyz_sheet5_df["Projected Ultimate"].loc[pd.Timestamp("2008-12-31")],
    46005,
    rtol=1e-3,
)
# Reconciles continuous calculation 514,181 and matches textbook 513,944 within 1e-3
assert np.isclose(xyz_sheet5_df["Projected Ultimate"].sum(), 514181, rtol=1e-3)
assert np.isclose(xyz_sheet5_df["Projected Ultimate"].sum(), 513944, rtol=1e-3)

# Exhibit III: Approach #2
assert np.isclose(exhibit_iii_df.loc[1998, "Case CDF (5)"], 1.506, atol=1e-3)
assert np.isclose(exhibit_iii_df.loc[2003, "Case CDF (5)"], 1.545, atol=1e-3)
assert np.isclose(exhibit_iii_df["Unpaid Estimate (6)"].sum(), 7011175, rtol=1e-3)

print("All reconciliation assertions successfully passed!")

All reconciliation assertions successfully passed!
